tgui/packages/tgui/interfaces/Smes.js

```diff
@@ -0,0 +1,207 @@
+/**
+ * @file
+ * @copyright 2020 Aleksej Komarov
+ * @author Original spookydonut (https://github.com/spookydonut)
+ * @author Changes Aleksej Komarov (https://github.com/stylemistake)
+ * @author Changes Sovexe (https://github.com/Sovexe)
+ * @license MIT
+ */
+
+import { useBackend } from '../backend';
+import { Box, Button, Flex, LabeledList, ProgressBar, Section, Slider } from '../components';
+import { formatPower } from '../format';
+import { Window } from '../layouts';
+
+// Common power multiplier
+const POWER_MUL = 1e3;
+
+export const Smes = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    capacityPercent,
+    charge,
+    formattedCharge,
+    inputAttempt,
+    inputting,
+    inputStartUp,
+    inputLevel,
+    inputLevelMax,
+    inputAvailable,
+    outputAttempt,
+    outputting,
+    outputLevel,
+    outputLevelMax,
+  } = data;
+  const inputState = (
+    capacityPercent >= 100 && 'good'
+    || (inputting && inputLevel) && 'average'
+    || 'bad'
+  );
+  const outputState = (
+    (outputAttempt && outputting) && 'good'
+    || charge > 0 && 'average'
+    || 'bad'
+  );
```

Comment:
```suggestion
  const inputState = (
    (capacityPercent >= 100 && 'good')
    || ((inputting && inputLevel) && 'average')
    || 'bad'
  );
  const outputState = (
    ((outputAttempt && outputting) && 'good')
    || (charge > 0 && 'average')
    || 'bad'
  );
```
Not technically required as I believe they will resolve correctly without parentheses, but makes intent much clearer.